File: 941-valid-mountain-array/941-valid-mountain-array.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        int n = arr.size();
        int i = 0, j = n - 1;
        while(i < n - 1 and arr[i] < arr[i + 1]){
            i += 1;
        }
        while(j > 0 and arr[j - 1] > arr[j]){
            j -= 1;
        }
        return (i == j and j < n - 1 and i > 0);
    }
};

File: 941-valid-mountain-array/941-valid-mountain-array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        n = len(arr)
        if n < 3 or arr[0] > arr[1] or arr[-1] > arr[-2]:
            return False
        i = 1
        while(arr[i] > arr[i - 1]):
            i += 1
        while i < n and arr[i - 1] > arr[i]:
            i += 1
        return i == n
        